### DISCUSSION TITLE
I'm not sure how I should move this to OM

### DISCUSSION THREAD

**Discussion by Aleeeeeek** [OpenMage/magento-lts#3884] [Questions & Answers]
I don't have a currrently running website but a backup of a magento 1.9.3.10 website (files folder and database). What is the best way to install openmage and migrate this magento backup into it?

**Comment by kiatng** [comment]
It really depends on what you intend to do. Take me as an example, I need to constantly add features and change behaviors of the shop. I  use git. For me, it's the best. I cannot live without it. (I know very little about git, I mainly use TortoiseGit instead of cli.)

If git is an overkill, then you can download the zip package and drop the files at the root. You need to consider which PHP ver you want. Which version to choose depends on your custom code and extensions. The more you have, the more incompatibility you need to fix. PHP7.4 is more backward compatible. PHP8.3 is the newest and fastest but has more incompatibility with old code.

**Reply by alexh-swdev** [reply]
Similar to kiatng:

I would do (I have done from a running system with some plugins) initially

1. get the backup of Magento 1.9.3.10 up and running.
2. put this into a git (I don't see any overhead there, it helps in so many ways...)
3. Use a file compare tool like FreeFileSync for windows (or BeyondCompare or ...) to merge the oldest available OM 19 version in
4. commit and push it when up and running

Then:
6. now you can use git to update to any newer OM version you want (latest 19.x and then 20.x) with the help of Git (or probably composer - I have not tried this yet, I am still doing it with git)

Sidenote: At least our 20.5.0 instances are running smooth with PHP8.3. (Just minor issues which are not related to the PHP version imho)

**Reply by kiatng** [reply]
Note on updating OM with git, after updating the OM to newer version, you need to run in terminal (PowerShell or bash or WSL2) `composer update` to update dependencies such as Zend.

**Reply by fballiano** [reply]
also, going from v19 to v20 you've a lot of stuff to remove: https://github.com/openmage/magento-lts/?tab=readme-ov-file#since-openmage-1950--2010

**Comment by addison74** [comment]
I would use a DDEV test environment to evaluate the migration to OpenMage.

1. I would make a file level comparison between Magento 1.9.3.10 and the file system you have. Araxis Merge (paid) is the best for Windows systems, it will show you the differences. Ideally, there should be no differences in the files related to Magento, that is, no one has directly changed the core files.

2. Delete all the PHP, JS, files common with Magento 1.9.3.10 distribution. Copy the OpenMage files. The custom changes will remain in places (/app/code/local, /app/code/community, /media, /skin/ ...).

3. Disable all custom extensions in /app/etc/modules. If you ran PHP 7.x, you definitely have the extension from Inchoo.

4. Load the Frontend/Backend in the browser and see what happens. You should have no issues.

5. Enable the logs in the Backend.

6. Enable the extensions one by one in /app/etc/modules and check the log files in /var/log. Most likely there will be deprecated errors that need to be fixed. Enable the Magento Developer mode to catch all errors.

7. Check all sections of the Frontend to make sure that the interface with the visitor is working properly.

8.  Evaluate further 1 - 2 weeks adding products, placing orders, as in a production environment. When you consider that everything is in order, keep the file system and use the initial database.
